src/Microsoft.ML.OnnxTransform/OnnxTransform.cs

```diff
@@ -0,0 +1,353 @@
+        {
+            private readonly IHost _host;
+            private readonly OnnxTransform _parent;
+
+            private readonly Type _outputItemRawType;
+            private readonly ColumnType _outputColType;
+            private readonly string _outputColName;
+
+            private readonly IdvToTensorAdapter _idvToTensorAdapter;
+
+            public Mapper(IHostEnvironment env, OnnxTransform parent, ISchema inputSchema)
+            {
+                Contracts.CheckValue(env, nameof(env));
+                _host = env.Register(nameof(Mapper));
+                _host.CheckValue(inputSchema, nameof(inputSchema));
+                _host.CheckValue(parent, nameof(parent));
+
+                _parent = parent;
+                var model = _parent.Model;
+                _idvToTensorAdapter = new IdvToTensorAdapter(inputSchema, parent._args.InputColumn,
+                                            model.ModelInfo.InputsInfo[0]);
+
+                // TODO: Remove assumption below
+                // Assume first output dimension is 1
+                var outputNodeInfo = model.ModelInfo.OutputsInfo[0];
+                int[] dims = outputNodeInfo.Shape.Skip(1).Select(x => (int)x).ToArray();
+                var outputItemType = OnnxUtils.OnnxToMlNetType(outputNodeInfo.Type);
+                _outputColType = new VectorType(outputItemType, dims);
+                _outputColName = _parent.Output;
+                _outputItemRawType = outputItemType.RawType;
+                _host.Assert(_outputItemRawType == _outputColType.ItemType.RawType);
+            }
+
+            public RowMapperColumnInfo[] GetOutputColumns()
+            {
+                var info = new RowMapperColumnInfo[1];
+                info[0] = new RowMapperColumnInfo(_outputColName, _outputColType, null);
+                return info;
+            }
+
+            public Func<int, bool> GetDependencies(Func<int, bool> activeOutput)
+            {
+                return col => activeOutput(0) && (_idvToTensorAdapter.IdvColumnIndex == col);
+            }
+
+            public void Save(ModelSaveContext ctx)
+            {
+                _parent.Save(ctx);
+            }
+
+            public Delegate[] CreateGetters(IRow input, Func<int, bool> activeOutput, out Action disposer)
+            {
+                disposer = null;
+                var getters = new Delegate[1];
+                using (var ch = _host.Start("CreateGetters"))
+                {
+                    if (activeOutput(0))
+                        getters[0] = Utils.MarshalInvoke(MakeGetter<int>, _outputItemRawType, input);
+
+                    ch.Done();
+                    return getters;
+                }
+            }
+
+            private Delegate MakeGetter<T>(IRow input)
+            {
+                _host.AssertValue(input);
+                _host.Assert(typeof(T) == _outputItemRawType);
+
+                ValueGetter<VBuffer<T>> valuegetter = (ref VBuffer<T> dst) =>
```

Comment:
>valuegetter [](start = 40, length = 11)

camelCase please

Reply:
Done.

---
In reply to: [220400214](https://github.com/dotnet/machinelearning/pull/942#discussion_r220400214) [](ancestors = 220400214)